Sources/SentryCrash/Recording/Tools/SentryCrashJSONCodecObjC.m

```diff
@@ -345,15 +345,28 @@ - (void)dealloc
     }
 
     if ([object isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *dict = (NSDictionary *)object;
         if ((result = sentrycrashjson_beginObject(context, cName)) != SentryCrashJSON_OK) {
             return result;
         }
-        NSArray *keys = [(NSDictionary *)object allKeys];
-        if (codec->_sorted) {
+        NSArray *keys = [dict allKeys];
+
+        bool allKeysOfSameType = true;
+        for (int i = 1; i < [keys count]; i++) {
+            if ([keys[i - 1] class] != [keys[i] class]) {
+                allKeysOfSameType = false;
+            }
+        }
+
+        // We can only sort the keys if all of them are of the same type, which is not guaranteed.
+        // Sorting an array with different types can cause a crash.
+        if (codec->_sorted && allKeysOfSameType) {
             keys = [keys sortedArrayUsingSelector:@selector(compare:)];
```

Comment:
M: I know that this code is already there, but why do we need to sort the array at all?

Reply:
It is an existing feature and some code might rely on it. I'm not 100% sure why it is there.

Reply:
I'd then check and doubt the need of having this sorting at all, just removing this would be enough to fix the issue (if sorting is not needed at all), it could be leftovers of v4... or not.

Reply:
At least [`SentryCrash.setUserInfo`](https://github.com/getsentry/sentry-cocoa/blob/854ddb4bca43cfd7c1119ec1bf0a9caa452bd74d/Sources/SentryCrash/Recording/SentryCrash.m#L166) uses this feature.

Reply:
I wonder because we don't do any sorting on Java/Android, pretty sure that our server doesn't rely on JSON key sorting.

Reply:
Okay, yes it would be weird if our server would rely on JSON key sorting. Still, we have a few two other places using this feature and therefore I would like to keep it.

Reply:
I'm fine keeping it as it is, but the question here needs to be answered just for the record.
do we really need this sorting and why? IMO this is part of the problem and to fix this issue in the best way, we need to answer that, either fixing the sorting as you did or removing it once for all.

Reply:
I am not 100% sure why KSCrash does this. Is this function only used when we call `setUserInfo`?
If that's the case I think it's safe to remove the sorting, if the sorting is used in other places as well we need to identify them and be careful.

Reply:
I would stick with it for now just to make sure.

Reply:
there's even a field to toggle this feature.

https://github.com/getsentry/sentry-cocoa/blob/12c03bd06287263984d69e2ea00f66f23ed3141e/Sources/SentryCrash/Recording/Tools/SentryCrashJSONCodecObjC.m#L60

I have no experience with the SentryCrash code, but based on these methods and docs, its just a feature/nice-to-have.

Reply:
So I would also stick to it. So can someone of you @marandaneto or @HazAT approve this PR?